.circleci/config.yml

```
@@ -68,85 +62,151 @@ jobs:
 
-      - run:
-          name: Build wheels
-          command: |
-            pyv=<< parameters.python-version >>
-            pyv="${pyv//.}"
-            pyv="${pyv:0:2}"
-            docker run -e pyv=$pyv --rm -v $PWD:/ws:rw --workdir=/ws \
-            quay.io/pypa/<< parameters.manylinux-tag >>_<< parameters.manylinux-arch >> \
-            bash -exc '
-              for py in /opt/python/cp*${pyv}*/bin/python;
-              do
-                  $py -m pip wheel . -w ./wheelhouse --no-deps;
-              done
-            '
-      - run:
-          name: bundle shared libraries into wheels
-          command: |
-            docker run -e pyv=$pyv --rm -v $PWD:/ws:rw --workdir=/ws \
-            quay.io/pypa/<< parameters.manylinux-tag >>_<< parameters.manylinux-arch >> \
-            bash -exc '
-              for whl in ./wheelhouse/dimod*.whl; do
-                auditwheel repair "$whl" -w ./dist
-              done
-            '
+    macos:
+      xcode: 12.5.1
 
-      - store_artifacts:
-          path: ./dist
+    environment:
+      <<: *global-environment
+      CIBW_PROJECT_REQUIRES_PYTHON: ~=<< parameters.python-version>>
+      CIBW_ARCHS_MACOS: << parameters.cibw-arch >>
 
-      - persist_to_workspace:
-          root: ./dist/
-          paths: .
+    steps: *build-steps
 
   build-sdist:
     docker:
-      - image: circleci/python:3.9-buster
-
-    working_directory: ~/build
+      - image: circleci/python:3.9
 
     steps:
       - checkout
-
       - run:
           name: build sdist
           command: |
-            python -m venv build_env
-            . build_env/bin/activate
+            python -m venv env
+            . env/bin/activate
             pip install -r requirements.txt
             python setup.py sdist -d ./dist
+      - store_artifacts: *store-artifacts
+      - persist_to_workspace: *persist-to-workspace
+
+  build-windows:
+    parameters:
+      python-version:
+        type: string
+
+    executor:
+      name: win/default
+
+    environment:
+      <<: *global-environment
+      CIBW_PROJECT_REQUIRES_PYTHON: ~=<< parameters.python-version>>
+      CIBW_ARCHS_WINDOWS: AMD64
+
+    steps:
+      - checkout
+      - run:
+          name: build wheels
+          command: |
+            python -m pip install cibuildwheel==2.2.2
+            python -m cibuildwheel --output-dir dist
+      - store_artifacts: *store-artifacts
+      - persist_to_workspace: *persist-to-workspace
+
+  deploy-all:
+    docker:
+      - image: circleci/python:3.9
+
+    steps:
+      - attach_workspace:
+          at: dist
 
       - store_artifacts:
           path: ./dist
 
-      - persist_to_workspace:
-          root: ~/build/dist/
-          paths: .
+      - run:
+          name: deploy
+          command: |
+            python -m venv env
+            . env/bin/activate
+            python -m pip install twine
+            twine upload -u $PYPI_USERNAME -p $PYPI_PASSWORD --skip-existing ./dist/*
```

```suggestion
            twine upload -u "$PYPI_USERNAME" -p "$PYPI_PASSWORD" --skip-existing ./dist/*
```
Password might contain unsafe chars (even if unlikely).